### PR TITLE
docs: Add more workflows to the GitHub Actions doc

### PR DESCRIPTION
**Pull request by Sherlock113**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [x] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: docs/source/data/bentoml-setup.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      - 'main'
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: bentoml/setup-bentoml-action@v1
+        with:
+          python-version: '3.10'
+          bentoml-version: 'main'
+          cache: 'pip'
```
Comment (aarnphm) [inline docs/source/data/bentoml-setup.yaml:14]: I dont' think this work yet. Let me update the implementation on setup-bentoml-action. Let's just remove this for now.
```suggestion
```

File: docs/source/data/containerize-and-push.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build Bento
+        run: bentoml build
+      - name: Build and push
+        uses: bentoml/containerize-push-action@v1
+        with:
+          bento-tag: my-bento:latest
+          push: true
+          tags: user/app:latest
```
Comment (aarnphm) [inline docs/source/data/containerize-and-push.yaml:31]:
```suggestion
      - uses: bentoml/build-bento-action@v1
        id: bento
      - name: Build and push BentoContainer
        uses: bentoml/containerize-push-action@v1
        with:
          bento-tag: ${{ steps.bento.outputs.bento-tag }}
          push: true
          tags: user/app:latest
```

File: docs/source/guides/github-actions.rst [hunk 3]
```diff
@@ -2,18 +2,99 @@
...
+     with:
+       bentofile: 'path/to/bentofile'
+
+.. note::
+
+   ``bentofile`` must be a path relative to the ``context`` directory.
+
+For more information, see the `build-bento-action repository <https://github.com/bentoml/build-bento-action>`_.
+
+Create and push Bento images to Docker Hub
```
Comment (aarnphm) [inline docs/source/guides/github-actions.rst:83]:
```suggestion
Create and push Bento images to Container registry
```

File: docs/source/guides/github-actions.rst [hunk 4]
```diff
@@ -2,18 +2,99 @@
...
+------------------------------------------
+
+You can create a GitHub Actions workflow to create and push Bento images to Docker Hub. This workflow uses `Docker Buildx <https://github.com/docker/buildx>`_,
+which offers extended capabilities for building containers, and is enhanced by the `Moby BuildKit builder toolkit <https://github.com/moby/buildkit>`_.
```
Comment (aarnphm) [inline docs/source/guides/github-actions.rst:87]:
```suggestion
You can create a GitHub Actions workflow to create and push Bento images toany container registry (Docker Hub, GHCR, ECR, GCR, etc.). This workflow uses `Docker Buildx <https://github.com/docker/buildx>`_,
which offers extended capabilities for building containers and is enhanced by the `Moby BuildKit builder toolkit <https://github.com/moby/buildkit>`_.
```

File: docs/source/guides/github-actions.rst [hunk 5]
```diff
@@ -2,18 +2,99 @@
...
+
+To use it, define a workflow file as below:
+
+.. literalinclude:: ./../data/containerize-and-push.yaml
+   :language: yaml
+
+This workflow first uses the `BentoML setup workflow <https://github.com/bentoml/setup-bentoml-action>`_ to prepare the necessary environment. It then
+sets up Docker QEMU and Docker Buildx, and configures Docker Hub authentication. Lastly, it builds a Bento, creates a Docker image, and pushes it to Docker Hub.
```
Comment (aarnphm) [inline docs/source/guides/github-actions.rst:95]:
```suggestion
In this example, the workflow first uses the `BentoML setup workflow <https://github.com/bentoml/setup-bentoml-action>`_ to prepare the necessary environment. It then
sets up Docker QEMU and Docker Buildx, and configures Docker Hub authentication. Lastly, it builds a Bento, creates a Docker image, and pushes it to Docker Hub.

.. note::

	You can also use ``docker/login-action@v2`` to login into other container registry that is supported by that action. Refer to https://github.com/docker/login-action?tab=readme-ov-file#usage for more information.
```

**Review by aarnphm — COMMENTED** [review]
Some comments

**Comment by Sherlock113** [comment]
@aarnphm Thanks. I made the changes accordingly.

**Review by aarnphm — APPROVED** [review]
LGTM. Thanks @Sherlock113

**Pull request merged**